pkg/frontend/v1/frontend.go

```diff
@@ -250,17 +250,24 @@ func getQuerierID(server frontendv1pb.Frontend_ProcessServer) (string, error) {
 }
 
 func (f *Frontend) queueRequest(ctx context.Context, req *request) error {
-	userID, err := tenant.TenantID(ctx)
+	tenantIDs, err := tenant.TenantIDs(ctx)
 	if err != nil {
 		return err
 	}
 
 	req.enqueueTime = time.Now()
 	req.queueSpan, _ = opentracing.StartSpanFromContext(ctx, "queued")
 
-	maxQueriers := f.limits.MaxQueriersPerUser(userID)
+	// figure out the highest max querier per user
+	var maxQueriers int
+	for _, tenantID := range tenantIDs {
+		v := f.limits.MaxQueriersPerUser(tenantID)
+		if v > maxQueriers {
```

Comment:
Should maxQueriers be the minimum value from all of the tenants being queried? Otherwise wouldn't it be possible to exceed the limit for a tenant with a low max queriers setting?

Reply:
Very good point, I was also missing the handling of the `0` and the same code in the query scheduler.

I have done that now in this commit: https://github.com/cortexproject/cortex/pull/3250/commits/3bc1204434e9988248ad14c07ef5f21f6fb756a9